src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,245 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, throwable);
+        }
+    }
+
+    private void insertDriversIntoCar(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertCarsDriversStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            insertCarsDriversStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertCarsDriversStatement.setLong(2, driver.getId());
+                insertCarsDriversStatement.executeUpdate();
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't insert drivers to car "
+                    + car + ". ", throwable);
+        }
+    }
+
+    private void removeAllDriversByCar(Car car) {
+        String query = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement removeCarsDriversStatement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            removeCarsDriversStatement.setLong(1, car.getId());
+            removeCarsDriversStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't remove drivers from car "
+                    + car + ". ", throwable);
+        }
+    }
+
+    private Set<Driver> getAllDriversByCarId(Long carId) {
+        String query = "SELECT cars.id AS car_id, "
+                + "drivers.id AS driver_id,  "
+                + "drivers.name AS driver_name,  "
+                + "drivers.license_number AS driver_license "
+                + "FROM cars "
+                + "JOIN cars_drivers "
+                + "ON cars_drivers.car_id = cars.id "
+                + "JOIN drivers "
+                + "ON cars_drivers.driver_id = drivers.id "
+                + "WHERE cars.id = ? AND cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversStatement = connection.prepareStatement(query)) {
+            getDriversStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversStatement.executeQuery();
+            Set<Driver> drivers = new HashSet<>();
+            while (resultSet.next()) {
+                Long driverCarId = resultSet.getObject("car_id", Long.class);
```

Comment:
Why do we need to get `car_id` from db if we pass it as input parameter?

Reply:
Fixed.